lib/ansible/modules/crypto/entrust/ecs_domain.py

```diff
@@ -0,0 +1,409 @@
+      the verification_method will be updated and validation data (if applicable) will be returned.
+    - If the domain is an active, validated domain, the return value of I(changed) will be false, unless C(domain_status=EXPIRED), in which case a re-validation
+      will be performed.
+    - If C(verification_method=DNS), details about the required DNS entry will be specified in the return parameters I(dns_contents), I(dns_location), and
+      I(dns_resource_type).
+    - If C(verification_method=WEB_SERVER), details about the required file details will be specified in the return parameters I(file_contents) and
+      I(file_location).
+    - If C(verification_method=EMAIL), the email address(es) that the validation email(s) were sent to will be in the return parameter I(emails). This is
+      purely informational. For domains requested using this module, this will always be a list of size 1.
+notes:
+    - There is a small delay (typically about 5 seconds, but as long as 60) before obtaining the random values when requesting a validation
+      while C(verification_method=DNS) or C(verification_method=WEB_SERVER). Be aware of that if doing many domain validation requests.
+options:
+    client_id:
+        description:
+            - The client ID to request the domain be associated with.
+            - If no client ID is specified, the domain will be added under the primary client with ID of 1.
+        type: int
+        default: 1
+    domain_name:
+        description:
+            - The domain name to be verified or reverified.
+        type: str
+        required: true
+    verification_method:
+        description:
+            - The verification method to be used to prove control of the domain.
+            - If C(verification_method=EMAIL) and the value I(verification_email) is specified, that value is used for the email validation. If
+              I(verification_email) is not provided, the first value present in WHOIS data will be used. An email will be sent to the address in
+              I(verification_email) with instructions on how to verify control of the domain.
+            - If C(verification_method=DNS), the value I(dns_contents) must be stored in location I(dns_location), with a DNS record type of
+              I(verification_DNS_record_type). To prove domain ownership, update your DNS records so the text string returned by I(dns_contents) is available at
+              I(DNS_location).
+            - If C(verification_method=WEB_SERVER), the contents of return value I(file_contents) must be made available on a web server accessible at location
+              I(file_location).
+            - If C(verification_method=MANUAL), the domain will be validated with a manual process. This is not recommended.
+        type: str
+        choices: [ 'DNS', 'EMAIL', 'MANUAL', 'WEB_SERVER']
```

Comment:
How about lower-case?
```suggestion
        choices: [ 'dns', 'email', 'manual', 'webserver']
```

Reply:
I was wary to deviate too much from the API spec that customers might be using elsewhere
>     enum:
>       - DNS
>       - EMAIL
>       - MANUAL
>       - WEB_SERVER

I'll change to lowercase.

Reply:
This is always a tough question, and I'm not sure what's the best solution :) In Ansible, things are usually lower-case. I guess you have to decide what you want in the end :)